components/SubmitPageForm/SubmitPageForm.tsx

```diff
@@ -0,0 +1,295 @@
+import React, { useState } from 'react';
+import Image from 'next/image';
+import { AiFillCloseCircle } from 'react-icons/ai';
+import {
+  useForm,
+  SubmitHandler,
+  Controller,
+  ControllerRenderProps,
+} from 'react-hook-form';
+import { useSession } from 'next-auth/react';
+import Router from 'next/router';
+import toast from 'react-hot-toast';
+import Lottie from 'lottie-react';
+import Loader from '../../public/loading.json';
+import { v4 as uuidv4 } from 'uuid';
+import axios from 'axios';
+import { Input } from '@/components/Form/Input';
+import { RichTextEditor } from '@/components/Form/RichTextEditor';
+import { FileDrop } from '@/components/Form/FileDrop';
+import { schema } from './schema';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+type FormInputs = {
+  tags: string[];
+  projectName: string;
+  repositoryLink: string;
+  projectDescription: string;
+  coverImage: string | null;
+  content: string | null;
+};
+
+const fileTypes: string[] = ['JPG', 'JPEG', 'PNG', 'GIF'];
+
+export const SubmitPageForm = () => {
+  const { handleSubmit, setValue, watch, control, reset } = useForm<FormInputs>(
+    {
+      resolver: yupResolver(schema),
+    }
+  );
+  const coverImageValue = watch('coverImage');
+  const tags = watch('tags');
+
+  const [tagInput, setTagInput] = useState<string>('');
+  const [fileError, setFileError] = useState(false);
+  const [fileKey, setFileKey] = useState(uuidv4().toString());
+
+  const { status, data: session } = useSession({
+    required: true,
+    onUnauthenticated() {
+      Router.push('/');
+    },
+  });
+
+  if (status === 'loading') {
+    return (
+      <div className="flex h-screen items-center justify-center">
+        <Lottie animationData={Loader} />
+      </div>
+    );
+  }
+
+  const onSubmit: SubmitHandler<FormInputs> = async (data: FormInputs) => {
+    try {
+      await axios.post(
+        '/api/project',
+        {
+          title: data.projectName,
+          description: data.projectDescription,
+          githubRepository: data.repositoryLink,
+          tags: data.tags,
+          coverImg:
+            'https://user-images.githubusercontent.com/48400770/190438248-fc0f3e42-c6d3-4d07-bcba-10e7fece4bc2.png',
+          email: session?.user?.email,
+        },
+        {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        }
+      );
+      toast.success('project added successfully');
+      reset();
+      setFileError(false);
+      setFileKey(uuidv4().toString());
+    } catch (e) {
+      toast.error(e.message);
+    }
+  };
+
+  const onTagAddition = (
```

Comment:
The text format should consistent.

![image](https://user-images.githubusercontent.com/60838584/196054279-39f15599-9fb9-4140-96b4-10cb772cb91e.png)

Reply:
okay 👍

Reply:
done